addons/mail/static/src/discuss/core/web/thread_actions.js

```diff
@@ -2,6 +2,22 @@ import { registerThreadAction } from "@mail/core/common/thread_actions";
 
 import { _t } from "@web/core/l10n/translation";
 
+export const joinChannelAction = {
+    condition: ({ owner, thread }) =>
+        thread &&
+        !["chat", "group"].includes(thread.channel_type) &&
+        !thread.self_member_id &&
+        !owner.isDiscussSidebarChannelActions,
```

Comment:
Why `!owner.isDiscussSidebarChannelActions`? Could be useful to have it over there.